qemu/tests/qmp_command.py

```diff
@@ -185,38 +185,21 @@ def check_result(qmp_o, output=None, exception_list=""):
                 raise exceptions.TestFail(msg)
 
     def qmp_cpu_check(output):
-        """ qmp_cpu test check """
-        last_cpu = int(params['smp']) - 1
-        for out in output:
-            cpu = out.get('CPU')
-            if cpu is None:
-                raise exceptions.TestFail("'CPU' index is missing in QMP output "
-                                          "'%s'" % out)
-            else:
-                current = out.get('current')
-                if current is None:
-                    raise exceptions.TestFail("'current' key is missing in QMP "
-                                              "output '%s'" % out)
-                elif cpu < last_cpu:
-                    if current is False:
-                        pass
-                    else:
-                        raise exceptions.TestFail("Attribute 'current' should be "
-                                                  "'False', but is '%s' instead.\n"
-                                                  "'%s'" % (current, out))
-                elif cpu == last_cpu:
-                    if current is True:
-                        pass
-                    else:
-                        raise exceptions.TestFail("Attribute 'current' should be "
-                                                  "'True', but is '%s' instead.\n"
-                                                  "'%s'" % (current, out))
-                elif cpu <= last_cpu:
-                    continue
-                else:
-                    raise exceptions.TestFail("Incorrect CPU index '%s' (corrupted "
-                                              "or higher than no_cpus).\n%s"
-                                              % (cpu, out))
+
+        """
+        qmp_cpu test check
+        :param output: return value of qmp command
+        :type output: string
+        """
+        cpu_num_cmd = int(params['smp'])
+        if output is None:
+            test.fail("No output from query-cpus command")
+        else:
+            cpu_num_guest = len(output)
+            if cpu_num_guest != cpu_num_cmd:
+                test.fail("'cpu num of guest' should be "
+                          "'%d', but is '%d' instead.\n"
```

Comment:
Escape character '\n' is not needed here.
I think it's more clear as following:
test.fail("guest cpu numbers is %d but specified smp is %d."
              % (cpu_num_guest, cpu_num_cmd))